src/utils/client.js

```diff
@@ -215,14 +220,18 @@ export default class Client {
     });
   }
 
-  submitCustom(customKeyValues) {
+  submitCustom(customKeyValues, signatureRef) {
     const { userKey } = initInfo();
     const url = `${API_URL}panel/custom/submit/v2/`;
 
     const jsonKeyVals = {};
     const formData = new FormData();
     Object.entries(customKeyValues).forEach(([key, val]) => {
-      if (val instanceof Blob) {
+      if (isBase64PNG(val)) {
+        if (signatureRef[key]) signatureRef[key].fromDataURL(val);
```

Comment:
there is also a submitCustom in commonCallbackProps that needs to be updated i believe. i think what may be cleaner is to put the signature update logic in updateFieldValues instead, that way it doesn't need to live in both updateSessionValues and submitCustom